docs/modules/ROOT/pages/erc20.adoc

```diff
@@ -0,0 +1,74 @@
+= ERC-20
+
+An ERC-20 token contract keeps track of xref:tokens.adoc#different-kinds-of-tokens[_fungible_ tokens]: any token is exactly equal to any other token; no token has a special right or behavior associated with them.
+This makes ERC-20 tokens useful for things like a *medium of exchange currency*, *voting rights*, *staking*, and more.
+
+OpenZeppelin Contracts provide many ERC20-related contracts for Arbitrum Stylus.
+On the https://docs.rs/openzeppelin_stylus/token/erc20/struct.Erc20.html[`API reference`] you'll find detailed information on their properties and usage.
+
+[[constructing-an-erc20-token-contract]]
+== Constructing an ERC-20 Token Contract
+
+Using Contracts, we can easily create our own ERC-20 token contract, which will be used to track _Gold_ (GLD), an internal currency in a hypothetical game.
+
+Here's what our GLD token might look like.
+
+[source,rust]
+----
+sol_storage! {
+    #[entrypoint]
+    struct GLDToken {
+        #[borrow]
+        Erc20 erc20;
+        #[borrow]
+        Erc20Metadata metadata;
+    }
+}
+
+#[external]
+#[inherit(Erc20, Erc20Metadata)]
+impl GLDToken {}
+----
+
+Our contracts are often used via stylus-sdk https://docs.arbitrum.io/stylus/reference/rust-sdk-guide#inheritance-inherit-and-borrow[inheritance], and here we're reusing https://docs.rs/openzeppelin_stylus/token/erc20/struct.Erc20.html[`ERC20`] for both the basic standard implementation and with optional extensions.
+
+// TODO: Add a section on how to deploy and call this contract
```

Comment:
TODO.

Reply:
I'm basically still thinking: we can make this section generic over the other contracts and have a separate page for deployment. Or put an example of deployment inside each contract..

Reply:
yeah, we should have one generic page about deployments, and then if anything is different for a certain contract, we explain it on its page.

Reply:
Created issues #227 and #226